### ISSUE TITLE
Extend NSDate such that it behaves like DateInRegion for the default or local region

### ISSUE THREAD

**Issue by Hout** [malcommac/SwiftDate#95]
Write the NSDate extension such that it behaves like DateInRegion for the local or default region. E.g. 
```
nsdate = NSDate()
regionDate = nsdate.inRegion(defaultRegion)

// Now nsdate and regionDate would behave equivalent
```

This would make the library easier to use if no conversions to other locales, time zones or calendard are necessary.
It would also make migration from SwiftDate 1.2 easier.

**Comment by redent** [comment]
I agree, but adding `inDefaultRegion()` to isn't that bad anyway.

**Comment by Hout** [comment]
Fixed in v3.0.0
